### PR TITLE
Large upload warning

### LINKED ISSUE #1506: feat: Warn users when they've uploaded something bigger than 1 MB
If users upload or record something bigger than 1 MB, everything looks and behaves well on Language Forge, but the audio will not make it into FLEx. 

We should warn users after they upload something larger than 1 MB that their file will not make it into FLEx so that they can consider adding a smaller file instead.

### PR DESCRIPTION
**Pull request by laineyhm**
Fixes #1506 

## Description

If users record audio or upload audio/images larger than 1MB/file, the files will save in Language Forge but will not sync with FLEx. This PR adds a warning to let users know when they've stored a file larger than 1MB so that they can consider using a smaller file if desired.

The first commit dealt with warnings for recorded and uploaded audio. The second commit dealt with warnings for uploaded image files.

### Type of Change

- UI change

## Screenshots

**Audio**
image

**Image**
image

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

## How to test

- [ ] Upload or record an audio file that is larger than 1MB. Notice whether the warning pops up at the top of the screen. The audio should still upload successfully to LF.
- [ ] Upload an image file that is larger than 1MB. Notice whether the warning pops up at the top of the screen. The image should still upload successfully to LF (unless it is a whole lot larger than 1MB).

## qa.languageforge.org testing

Testers: Check the box and put in a date/time to sign-off/attest the feature works as expected on qa.languageforge.org

- [ ] Tester1 (YYYY-MM-DD HH:MM)
- [ ] Tester2 (YYYY-MM-DD HH:MM)

### CODE REVIEW

File: src/angular-app/languageforge/lexicon/editor/field/dc-audio.component.ts [hunk 1]
```diff
@@ -115,6 +115,9 @@ export class FieldAudioController implements angular.IController {
             this.dcFilename = response.data.data.fileName;
             this.showAudioUpload = false;
             this.notice.push(this.notice.SUCCESS, 'File uploaded successfully.');
+            if(response.data.data.fileSize > 1000000){ //1 MB file size limit 2022-09
```
Comment (megahirt) [inline src/angular-app/languageforge/lexicon/editor/field/dc-audio.component.ts:118]: I'm good with 1 million bytes here, even though a megabyte likely refers to [1024 x 1024 = 1,048,576 bytes](https://www.backblaze.com/blog/how-many-bytes-are-in-a-megabyte-really/).  I still get confused by this and am not 100% sure which one the Chorus 1MB limit is really referring to.

File: src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php [hunk 2]
```diff
@@ -125,11 +126,16 @@ public static function uploadAudioFile($projectId, $mediaType, $tmpFilePath)
 
             }
 
+            //Find the size of the upload file now pointed to by the entry
+            $fileSize = intval(`ffprobe -i $filePath -show_entries format=size -v quiet -of csv="p=0"`);
```
Comment (megahirt) [inline src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php:130]: While this works, using ffprobe to get the file size of a file is probably overkill.  How about this?
```suggestion
            $fileSize = intval(`ls -l $filePath | awk '{print $5}'`);
```

**Review by megahirt — CHANGES_REQUESTED** [review]
Another great PR!  One small requested change regarding using `ls` instead of `ffprobe` for getting the file size.

**Comment by alex-larkin** [comment]
This warning is a great idea! Good work. 🙌

File: src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php (hunk 2 above)
Comment (laineyhm) [reply src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php]: In the latest commit I use the php filesize command for both audio and images.

Comment (megahirt) [reply src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php]: thank you

File: src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php [hunk 3]
```diff
@@ -125,11 +126,16 @@ public static function uploadAudioFile($projectId, $mediaType, $tmpFilePath)
...
+            //Find the size of the upload file now pointed to by the entry
+            $fileSize = filesize($filePath);
+
+
             // construct server response
             if ($moveOk && $tmpFilePath) {
                 $data = new MediaResult();
                 $data->path = $project->getAudioFolderPath($project->getAssetsRelativePath());
                 $data->fileName = $fileNamePrefix . '_' . $fileName; //if the file has been converted, $fileName = converted file
+                $data->fileSize = $fileSize;
```
Comment (megahirt) [inline src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php:138]: Now that calculating the filesize is trivial, let's do it all on one line
```suggestion
                $data->fileSize = filesize($filePath);
```

File: src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php [hunk 4]
```diff
@@ -125,11 +126,16 @@ public static function uploadAudioFile($projectId, $mediaType, $tmpFilePath)
 
             }
 
+            //Find the size of the upload file now pointed to by the entry
+            $fileSize = filesize($filePath);
+
+
```
Comment (megahirt) [inline src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php:132]:
```suggestion
```

File: src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php [hunk 5]
```diff
@@ -221,11 +227,15 @@ public static function uploadImageFile($projectId, $mediaType, $tmpFilePath)
             $moveOk = copy($tmpFilePath, $filePath);
             @unlink($tmpFilePath);
 
+            //Find the size of the image file
+            $fileSize = filesize($filePath);
+
```
Comment (megahirt) [inline src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php:232]:
```suggestion
```

File: src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php [hunk 6]
```diff
@@ -221,11 +227,15 @@ public static function uploadImageFile($projectId, $mediaType, $tmpFilePath)
...
             // construct server response
             if ($moveOk && $tmpFilePath) {
                 $data = new MediaResult();
                 $data->path = $project->getImageFolderPath($project->getAssetsRelativePath());
                 $data->fileName = $fileNamePrefix . '_' . $fileName;
+                $data->fileSize = $fileSize;
```
Comment (megahirt) [inline src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php:238]:
```suggestion
                $data->fileSize = filesize($filePath);
```

**Review by megahirt — CHANGES_REQUESTED** [review]
A small nit picky change to make it tidier.

**Review by megahirt — APPROVED** [review]
LGTM

**Pull request merged**